### PR TITLE
Implement dummy runner and basic instance runner

### PR DESCRIPTION
**Pull request by fishnix**
This mainly implements the basic `dummyRunner` and `instanceRunner` and the corresponding configuration wiring.  It solidifies the interface for a runner as well.  The endpoint for executing a job exists currently because the scheduler doesn't.  That `PATCH` endpoint will probably change or be removed once the scheduler is in place (although I could see a case where someone may want to test their job execution...).

### CODE REVIEW

**Comment by damnski** [comment]
:+1:

File: jobs/instancerunner_test.go [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+	out, err = instanceRunnerTmpl.Run(context.TODO(), "myaccount", map[string]string{
+		"instance_id":     "i-123456",
+		"instance_action": "stop",
+	})
+	if err != nil {
+		t.Errorf("expected nil error, got %s", err)
+	}
+
+	if out != "KO" {
+		t.Errorf("expected OK, got %s", out)
```
Comment (tenyo) [inline jobs/instancerunner_test.go]: 👀 `KO`

File: jobs/instancerunner_test.go [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+	if _, err = instanceRunner.Run(context.TODO(), "", "foo"); err == nil {
+		t.Error("expected error for empty account, got nil")
+	}
+
+	if _, err = instanceRunner.Run(context.TODO(), "", map[string]string{}); err == nil {
+		t.Error("expected error for missing instance_id, got nil")
+	}
+
+	if _, err = instanceRunner.Run(context.TODO(), "", map[string]interface{}{
+		"instance_id": []string{"foo", "bar"},
```
Comment (tenyo) [inline jobs/instancerunner_test.go]: These tests have an empty account, so they are working (failing) but for the wrong reason, maybe we need to check the specific error message

File: jobs/instancerunner.go [hunk 3]
```diff
@@ -120,37 +124,23 @@ func (r *InstanceRunner) Run(ctx context.Context, account string, parameters int
...
 		if err != nil {
-			return "", err
+			return "", NewRunnerError(ErrPostExecFailure, "reading response body failed", err)
 		}
 
 		log.Debugf("got response %s(%d) for endpoint %s: %s", res.Status, res.StatusCode, endpoint, body)
 
 		if res.StatusCode >= 300 {
-			return "", errors.New("unexpected response from instanceRunner api: " + res.Status)
+			return "", NewRunnerError(ErrExecFailure, "unexpected http respons", errors.New("unexpected response from instanceRunner api: "+res.Status))
```
Comment (tenyo) [inline jobs/instancerunner.go]: 👀 `respons`

**Pull request merged**